qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -0,0 +1,176 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy experiment class for resonators."""
+
+from typing import Iterable, Optional, Tuple
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Backend
+import qiskit.pulse as pulse
+
+from qiskit_experiments.library.characterization import QubitSpectroscopy
+from qiskit_experiments.data_processing.processor_library import get_processor, ProjectorType
+from .analysis.resonator_spectroscopy_analysis import ResonatorSpectroscopyAnalysis
+
+
+class ResonatorSpectroscopy(QubitSpectroscopy):
```

Comment:
That true. See here: bde77fb

Reply:
Yes, I think it would even be better to take on some redundancy for the sake of clarity because this is a fairly simple experiment and it would be nice if the class were readable without needing to understand what was inherited from qubit spectroscopy.

Also, I think right now this is producing a `f01` result for component `Q1`? In https://github.com/Qiskit/qiskit-experiments/pull/387, it had been suggested that this experiment might be the one to introduce resonator components? We have used separate resonator components in the past. Personally, I don't mind the resonator properties being associated with the qubit component but we should double check with other users. Also, `ResonanceAnalysis` currently fits to a gaussian where for a resonator a Lorentzian fit would be more appropriate.